### ISSUE TITLE
How to align Dropdown inside Navbar with collapse? 

### ISSUE THREAD

**Issue by shobhit-shrivastava** [bestguy/sveltestrap#143]
I am new to sveltestrap so forgive me if this is a dumb question.
I followed this sample code from the storybook.  https://bestguy.github.io/sveltestrap/?path=/story/components--navbar
This example uses Uncontrolleddropdown inside a Navbar. The dropdown aligns to the left or right of the page. I want the dropdown to be below the dropdownmenu. I tried replacing uncontrolleddropdown with a regular dropdown and I added a toggle function to regular dropdown but it doesn't seem to work.
Also, if i remove the navbar the regular dropdown with toggle function works as expected.
Could you please show me how to use a regular dropdown inside a navbar ? 
I have been scratching my head trying to find a solution so any help is greatly appreciated.
```
<script>
  import {
    Collapse,
    Navbar,
    NavbarToggler,
    NavbarBrand,
    Nav,
    NavItem,
    NavLink,
    UncontrolledDropdown,
    DropdownToggle,
    DropdownMenu,
    DropdownItem
  } from 'sveltestrap';

  let isOpen = false;

  function handleUpdate(event) {
    isOpen = event.detail.isOpen;
  }
</script>

<Navbar color="light" light expand="md">
  <NavbarBrand href="/">sveltestrap</NavbarBrand>
  <NavbarToggler on:click={() => (isOpen = !isOpen)} />
  <Collapse {isOpen} navbar expand="md" on:update={handleUpdate}>
    <Nav class="ml-auto" navbar>
      <NavItem>
        <NavLink href="#components/">Components</NavLink>
      </NavItem>
      <NavItem>
        <NavLink href="https://github.com/bestguy/sveltestrap">GitHub</NavLink>
      </NavItem>
      <UncontrolledDropdown nav inNavbar>
        <DropdownToggle nav caret>Options</DropdownToggle>
        <DropdownMenu right>
          <DropdownItem>Option 1</DropdownItem>
          <DropdownItem>Option 2</DropdownItem>
          <DropdownItem divider />
          <DropdownItem>Reset</DropdownItem>
        </DropdownMenu>
      </UncontrolledDropdown>
    </Nav>
  </Collapse>
</Navbar>
```

**Comment by NickKaramoff** [comment]
I believe this is a bug. The same behaviour is also present on my page without collapsing navbar:

```svelte
<Navbar color="dark" dark expand class="d-flex justify-content-between">
	<Nav navbar>
		<!-- other nav elements -->
	    <UncontrolledDropdown nav inNavbar>
		    <DropdownToggle caret nav>
		        <Icon name="arrow-up-down" />
		        Sort
		    </DropdownToggle>
		    <DropdownMenu right>
		        {#each sortTypes as st}
		            <DropdownItem>{st.name}</DropdownItem>
		        {/each}
		    </DropdownMenu>
		</UncontrolledDropdown>
	</Nav>
</Navbar>
```

![Image of the bug](https://user-images.githubusercontent.com/6566248/79148295-8d3f8100-7dc5-11ea-9b84-ce34c5552fdb.png)

In my example, the dropdown menu is more or less near the toggle, but it is still wrong — it aligns to the page and not to the toggle.

**Comment by NickKaramoff** [comment]
I was able to find the source of the problem.

The code

```svelte
<Navbar color="dark" dark expand class="d-flex justify-content-between">
	<Nav navbar>
		<!-- other nav elements -->
	    <UncontrolledDropdown nav inNavbar>
		    <!-- dropdown content -->
		</UncontrolledDropdown>
	</Nav>
</Navbar>
```

is being rendered to (Svelte attributes stripped out)

```html
<nav class="d-flex justify-content-between navbar navbar-expand navbar-dark bg-dark">
	<ul class="navbar-nav">
		<!-- other nav elements -->
		<li class="">
		    <!-- dropdown content -->
		</li>
	</ul>
</nav>
```

As you can see, the `<li>` representing the dropdown doesn't have required classes `nav-item` and `dropdown`, although they should be present according to https://github.com/bestguy/sveltestrap/blob/88c1c30d8086768c4f9e4bce2a9fc9cf31920d45/src/Dropdown.svelte#L53 and https://github.com/bestguy/sveltestrap/blob/88c1c30d8086768c4f9e4bce2a9fc9cf31920d45/src/Dropdown.svelte#L55

I tried to debug the component but I still couldn't find the reason why this problem appears :/

**Comment by NickKaramoff** [comment]
@shobhit-shrivastava the permanent solution I came up with for now is to add `dropdown` class to the `<UncontrolledDropdown>`/`<Dropdown>` element:

```svelte
<UncontrolledDropdown nav inNavbar class="dropdown">
    <!-- the rest of the dropdown -->
</UncontrolledDropdown>
```

**Comment by xapdkop** [comment]
This issue is the incorrect placement of spreading `props` in the [Dropdown.svelte](https://github.com/bestguy/sveltestrap/blob/88c1c30d8086768c4f9e4bce2a9fc9cf31920d45/src/Dropdown.svelte) (`props` should be spread before any other attributes since `props` can override them)
https://github.com/bestguy/sveltestrap/blob/88c1c30d8086768c4f9e4bce2a9fc9cf31920d45/src/Dropdown.svelte#L100
https://github.com/bestguy/sveltestrap/blob/88c1c30d8086768c4f9e4bce2a9fc9cf31920d45/src/Dropdown.svelte#L104

---

For example, when you're using `UncontrolledDropdown`, it "under the hood" creates `Dropdown` and sets `class` attribute to empty string [hmmmm, is this a good idea???].
https://github.com/bestguy/sveltestrap/blob/88c1c30d8086768c4f9e4bce2a9fc9cf31920d45/src/UncontrolledDropdown.svelte#L5
https://github.com/bestguy/sveltestrap/blob/88c1c30d8086768c4f9e4bce2a9fc9cf31920d45/src/UncontrolledDropdown.svelte#L27
After it in the `Dropdown` props from `UncontrolledDropdown` are being spread and empty `class` from them replaces `class={classes}`

**Comment by bestguy** [comment]
Closing, released in v3.4.0
